### PR TITLE
CSPL-1787 - Updated docs to add scc nonroot to splunk-operator-controller-manager in openshift environments

### PR DESCRIPTION
**Pull request by akondur**
The Splunk Operator pod is attached to the Service Account `splunk-operator-controller-manager` and runs as user `1001`. On deploying the operator in openshift environments, it throws the following error(tested on an openshift cluster):

```
Events:
  Type     Reason        Age                 From                   Message
  ----     ------        ----                ----                   -------
  Warning  FailedCreate  33s (x14 over 74s)  replicaset-controller  Error creating: pods "splunk-operator-controller-manager-6465874c68-" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, spec.containers[0].securityContext.runAsUser: Invalid val
ue: 1001: must be in the ranges: [1000580000, 1000589999], spec.containers[1].securityContext.runAsUser: Invalid value: 1001: must be in the ranges: [1000580000, 1000589999], provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "machin
e-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usa
ble by user or serviceaccount]
```

Updated documentation to add the `nonroot` scc to the serviceaccount for the operator.

Note: Once the nonroot scc is added, the pods were deployed properly:

```
$ kgp
NAME                                                  READY   STATUS    RESTARTS   AGE
splunk-operator-controller-manager-6465874c68-6r6sm   2/2     Running   0          22m
```

Fixing github [issue](https://github.com/splunk/splunk-operator/issues/764).

### CODE REVIEW

**Review by jryb — APPROVED** [review]

File: docs/OpenShift.md [hunk 1]
```diff
@@ -9,11 +9,15 @@ container and compromising the host. For more information, please see the
...
 [Documentation on Security](https://github.com/splunk/docker-splunk/blob/develop/docs/SECURITY.md).
 
+Similarly the Splunk Operator pod is attached to the Service Account
+`splunk-operator-controller-manager` and runs as user `1001`.
+
 Users of Red Hat OpenShift may find that the default Security Context
-Constraint is too restrictive. You can fix this by granting the default
-Service Account the `nonroot` Security Context Constraint by running the
-following command within your namespace:
+Constraint is too restrictive. You can fix this by granting the `default`
```
[the remark this answers is not in the document]
Comment (akondur) [reply docs/OpenShift.md:16]: The splunk enterprise deployments are attached to the default service account though right?

[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply docs/OpenShift.md:16]: I think user can pass service account in CRD if not given it uses default service account. so there are 2 issues here. one splunk-operator installation which was failing thats fixed by adding constraint to splunk-operator-controller-manager service account. another is splunk instances created by splunk-operator which may or may not use default serviceaccount.

[the remark this answers is not in the document]
Comment (akondur) [reply docs/OpenShift.md:16]: Thanks @vivekr-splunk , let me rephrase the same accordingly.

File: docs/OpenShift.md [hunk 2]
```diff
@@ -1,19 +1,29 @@
 # Red Hat OpenShift Configuration
 
 The Splunk Operator will always start Splunk Enterprise containers using
-a specific, unprivileged `splunk` user and group. Running as a specific user
-and group is required for the Splunk Enterprise containers to write to
-Kubernetes PersistentVolumes. This follows best security practices, which
-helps prevent any malicious actor from escalating access outside of the
+a specific, unprivileged `splunk(41812)` user and group to allow write access  to Kubernetes PersistentVolumes. This follows best security practices,
```
Comment (smohan-splunk) [inline docs/OpenShift.md:4]: Nit: An extra space between "access" & "to"

File: docs/OpenShift.md [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
-a specific, unprivileged `splunk` user and group. Running as a specific user
-and group is required for the Splunk Enterprise containers to write to
-Kubernetes PersistentVolumes. This follows best security practices, which
-helps prevent any malicious actor from escalating access outside of the
+a specific, unprivileged `splunk(41812)` user and group to allow write access  to Kubernetes PersistentVolumes. This follows best security practices, 
+which helps prevent any malicious actor from escalating access outside of the
 container and compromising the host. For more information, please see the
 Splunk Enterprise container's
 [Documentation on Security](https://github.com/splunk/docker-splunk/blob/develop/docs/SECURITY.md).
+The Splunk Enterprise pods are attached to the `default` serviceaccount or the configured
```
Comment (smohan-splunk) [inline docs/OpenShift.md:9]: Can this be part of the next paragraph, along with Operator pod's service account explanation?

**Review by smohan-splunk — APPROVED** [review]

**Review by vivekr-splunk — APPROVED** [review]
